src/ui/public/chrome/directives/append_nav_controls.js

```diff
@@ -17,16 +13,42 @@ export default function (chrome, internals) {
     return {
       template: function ($element) {
         const parts = [$element.html()];
-        const controls = Private(chromeNavControlsRegistry);
+        const navs = Private(chromeNavControlsRegistry);
+        const configs = Private(chromeConfigControlsRegistry);
+
+        const controls = [
+          {
+            name: 'active http requests',
+            order: -100,
+            template: spinnerHtml
+          },
+
+          ...navs.map(function (nav) {
+            return {
+              template: `<!-- nav control ${nav.name} -->${nav.template}`,
+              order: nav.order
+            };
+          }),
+
+          ...configs.map(function (config) {
+            const configHtml = `<render-directive definition="configs['${config.name}'].navbar">
```

Comment:
The JSON.stringify didn't work since it add double quotes which messed up the html parsing.  I instead add a regex piece in the registry that forces the names to only allow a-z, A-Z, or 0-9.